lib/OpenQA/CacheService/Task/Sync.pm

```diff
@@ -39,10 +41,17 @@ sub _cache_tests {
     my @cmd = (qw(rsync -avHP --timeout), RSYNC_TIMEOUT, "$from/", qw(--delete), "$to/tests/");
     my $cmd = join ' ', @cmd;
     $ctx->info("Calling: $cmd");
-    my $output = `@cmd`;
-    my $status = $? >> 8;
+    my $status;
+    my $full_output;
+    for my $retry (1 .. RSYNC_RETRIES) {
+        my $output = `@cmd`;
+        $status = $? >> 8;
+        $full_output = "Try $retry:\n" . $output;
+        last unless $status;
+        sleep RSYNC_RETRY_PERIOD;
```

Comment:
To cover this in tests without needles waiting I suggest to use https://metacpan.org/pod/Test::Mock::Time which we already use in os-autoinst. You just need to specify the package as dependency in openQA but it's already installed in the CI container due to the os-autoinst dependency anyway

Reply:
Or you simply set `OPENQA_RSYNC_RETRY_PERIOD=0` in the test environment. (Needs to be in a `BEGIN` block as it is a constant.)

Reply:
> Or you simply set `BEGIN { $ENV{OPENQA_RSYNC_RETRY_PERIOD} = 0 }` in the test environment. (Needs to be in a `BEGIN` block as it is a constant.)

tried, not working:( so time for Test::Mock::Time

Reply:
Strange, it should've been working - especially since there are already similar overrides in the same test file.

Reply:
and `Test::More::Time` broke *Subtest: Minion monitoring with InfluxDB* , so probably best option is mark `sleep` line as non-coverable :(

Reply:
So you would want to mark the line as non-coverable because you want to put in some extra condition to not run the code at all? I'm just asking because the issue we want to avoid is that tests take longer due to the sleep.